src/app/modules/auth/auth.service.ts

```diff
@@ -298,3 +299,25 @@ export const getFormAfterPermissionChecks = ({
     ),
   )
 }
+
+/**
+ * Retrieves the form of given formId provided that the form is public.
+ *
+ * @returns ok(form) if the form is public
+ * @returns err(FormNotFoundError) if form (or form admin) does not exist
+ * @returns err(FormDeletedError) if form is already archived
+ * @returns err(PrivateFormError) if form is private
+ * @returns err(DatabaseError) if database error occurs
+ */
+export const getFormIfPublic = (
+  formId: string,
+): ResultAsync<
+  IPopulatedForm,
+  FormNotFoundError | FormDeletedError | PrivateFormError | DatabaseError
+> => {
+  return FormService.retrieveFullFormById(formId).andThen((form) =>
+    FormService.isFormPublic(form, 'Cannot access private form').map(
```

Comment:
this `map` should be outside right?

Reply:
nope, since isFormPublic only returns `true` (or an error) (we should probably rename the function sometime in a different PR)

Reply:
I'll add a comment